### PR TITLE
Add release notes for 1.0.0-beta

### PR DESCRIPTION
**Pull request by aturon**
Collected from TWiR and the PR log.

r? @alexcrichton 
f? @brson @nikomatsakis @huonw

### CODE REVIEW

File: RELEASES.md [hunk 1]
```diff
@@ -1,3 +1,102 @@
...
+      and provide ergonomic [blanket implementations][fn-blanket].
+    * The `Index` and `IndexMut` traits were changed to
+      [take the index by value][index-value], enabling code like
+      `hash_map["string"]` to work.
+    * `Copy` now [inherits][copy-clone] from `Clone`, meaning that all
+      `Copy` data is known to be `Clone` as well.
+
+* Infrastructure
+
+    * Metadata was tunes, shrinking binaries [by 27%][metadata-shrink].
```
Comment (sfackler) [inline RELEASES.md]: tuned

**Comment by aturon** [comment]
f? @nrc @pnkfelix

**Comment by pnkfelix** [comment]
@aturon Assuming #23945 lands, then something needs to mention it, though I would be satisfied with just making it an amendment on https://github.com/rust-lang/rust/pull/22532

File: RELEASES.md [hunk 2]
```diff
@@ -1,3 +1,102 @@
...
+
+    * Metadata was tuned, shrinking binaries [by 27%][metadata-shrink].
+    * Much headway was made on ecosystem-wide CI, making it possible
+      to [compare builds for breakage][ci-compare].
+
+[send-rfc]: https://github.com/rust-lang/rfcs/blob/master/text/0458-send-improvements.md
+[scoped]: http://static.rust-lang.org/doc/master/std/thread/fn.scoped.html
+[moar-ufcs]: https://github.com/rust-lang/rust/pull/22172
+[prim-inherent]: https://github.com/rust-lang/rust/pull/23104
+[overflow]: https://github.com/rust-lang/rust/pull/22532
```
Comment (pnkfelix) [inline RELEASES.md]: should this point instead at the integer overflow RFC?  Or did you want to keep focus on what was actually implemented, versus what pieces still may remain unimplemented from the RFC?  (Or is the RFC PR itself not a useful document to point to ... in which  case I would suggest pointing to the text of the merged RFC itself?)

I'm just not sure the implementation PR is itself the most useful pointer to provide for this document.

File: RELEASES.md [hunk 3]
```diff
@@ -1,3 +1,102 @@
...
+      traits like `IteratorExt` to be merged into the traits they
+      extended.
+    * You can now [refer to associated types][assoc-where] whose
+      corresponding trait bounds appear only in a `where` clause.
+    * The final bits of [OIBIT landed][oibit-final], meaning that
+      traits like `Send` and `Sync` are now library-defined.
+    * A [Reflect trait][reflect] was introduced, which means that
+      downcasting via the `Any` trait is effectively limited to
+      concrete types. This helps retain the potentially-important
+      "parametricity" property: generic code cannot behave differently
```
Comment (pnkfelix) [inline RELEASES.md]: > This helps retain the potentially-important "parametricity" property: generic code cannot behave differently for different type arguments.

You may want to weaken this claim slightly.  For example, there's still the option of different behavior based on the value of `mem::size_of::<T>()`, among other options.  Up to you.

**Comment by pnkfelix** [comment]
@aturon @alexcrichton okay, r=me whenever you want to put this in.

**Comment by aturon** [comment]
@pnkfelix Thanks for the tweaks (I put this together in a hurry)!

@alexcrichton Feel free to update with all of @pnkfelix's suggestions.

**Pull request merged**